src/TestProblem/Hydro/CCSN/Flag_CCSN.cpp

```diff
@@ -34,46 +39,44 @@ extern double CCSN_CentralDens;
 bool Flag_CoreCollapse( const int i, const int j, const int k, const int lv, const int PID, const double *Threshold )
 {
 
-   bool Flag = false;
+   bool Flag      = false;
    bool MaxRefine = false;
 
    const double dh        = amr->dh[lv];
-   const double Center[3] = { amr->BoxCenter[0], amr->BoxCenter[1], amr->BoxCenter[2] };
    const double Pos   [3] = { amr->patch[0][lv][PID]->EdgeL[0] + (i+0.5)*dh,
                               amr->patch[0][lv][PID]->EdgeL[1] + (j+0.5)*dh,
                               amr->patch[0][lv][PID]->EdgeL[2] + (k+0.5)*dh  };
-
-   const double dx = Center[0] - Pos[0];
-   const double dy = Center[1] - Pos[1];
-   const double dz = Center[2] - Pos[2];
-   const double r  = sqrt(  SQR( dx ) + SQR( dy ) + SQR( dz )  );
+   const double Center[3] = { 0.5*amr->BoxSize[0], 0.5*amr->BoxSize[1], 0.5*amr->BoxSize[2] };
+   const double dR[3]     = { Pos[0]-Center[0], Pos[1]-Center[1], Pos[2]-Center[2] };
+   const double R         = sqrt( SQR(dR[0]) + SQR(dR[1]) + SQR(dR[2]) );
 
    const double CentralDens = CCSN_CentralDens / UNIT_D;
 
 // (1) check if the allowed maximum level is reached
-   if ( CCSN_CC_MaxRefine_Flag1  &&  CentralDens < CCSN_CC_MaxRefine_Dens1 / UNIT_D )
-   {
+   if      ( CCSN_CC_MaxRefine_Flag1  &&  CentralDens < CCSN_CC_MaxRefine_Dens1 / UNIT_D )
       MaxRefine = lv >= CCSN_CC_MaxRefine_LV1;
-   }
 
    else if ( CCSN_CC_MaxRefine_Flag2  &&  CentralDens < CCSN_CC_MaxRefine_Dens2 / UNIT_D )
-   {
       MaxRefine = lv >= CCSN_CC_MaxRefine_LV2;
-   }
 
 
 // (2) always refined to highest level in the region with r < 30 km
    if ( !MaxRefine )
    {
 //    (2-a) always refine the innermost cells
-      if ( r < amr->dh[lv] )
+      if ( R < amr->dh[lv] )
          Flag = true;
 
 //    (2-b) refine the region with r < 30 km
-      if ( r * UNIT_L < 3e6 )
+      if ( R * UNIT_L < 3e6 )
          Flag = true;
    }
 
+// (3) check if the minimum angular resolution (in degree) is reached
+   if ( CCSN_Min_AngRes > 0.0  && !MaxRefine  &&  !Flag )
+      if ( R * CCSN_Min_AngRes < dh * RAD2DEG )
+         Flag = true;
```

Comment:
- move this into the `if ( !MaxRefine )` block above, and remove unnecessary conditions
- combine the two `if` statement
- use two spaces before and after `&&`

Reply:
- I moved and removed unnecessary conditions.
- I combined this.
- I also corrected the space.

Reply:
I think we can combine criteria (2) and (3) further:
``` c
if ( !MaxRefine )
{
//   (2) ...
//   (3) ...
}
```

Reply:
I combined this part.